rclcpp/include/rclcpp/time.hpp

```diff
@@ -44,31 +69,38 @@ class Time
       throw std::runtime_error("RCL_ROS_TIME is currently not implemented.");
     }
 
-    rcl_ret_t ret = RCL_RET_ERROR;
-
-    rcl_time_source_t time_source;
-    ret = rcl_time_source_init(ClockT, &time_source);
+    auto time_point = get_empty_time_point<ClockT>();
 
+    auto ret = rcl_time_point_get_now(&time_point);
     if (ret != RCL_RET_OK) {
       rclcpp::exceptions::throw_from_rcl_error(
-        ret, "could not initialize time source: ");
+        ret, "could not get current time stamp: ");
     }
 
-    rcl_time_point_t time_point;
-    ret = rcl_time_point_init(&time_point, &time_source);
+    return Time(std::move(time_point));
+  }
 
-    if (ret != RCL_RET_OK) {
-      rclcpp::exceptions::throw_from_rcl_error(
-        ret, "could not initialize time point: ");
-    }
+  template<rcl_time_source_type_t ClockT = RCL_SYSTEM_TIME>
+  Time(int32_t seconds, uint32_t nanoseconds)
+  : rcl_time_(get_empty_time_point())
+  {
+    uint64_t ns = RCL_S_TO_NS(seconds);
+    ns += nanoseconds;
+    rcl_time_.nanoseconds = ns;
+  }
 
-    ret = rcl_time_point_get_now(&time_point);
-    if (ret != RCL_RET_OK) {
-      rclcpp::exceptions::throw_from_rcl_error(
-        ret, "could not get current time stamp: ");
-    }
+  template<rcl_time_source_type_t ClockT = RCL_SYSTEM_TIME>
+  explicit Time(uint64_t nanoseconds)
+  : rcl_time_(get_empty_time_point())
+  {
+    rcl_time_.nanoseconds = nanoseconds;
+  }
 
-    return Time(std::move(time_point));
+  Time(const builtin_interfaces::msg::Time & time_msg)  // NOLINT
+  : rcl_time_(get_empty_time_point())
+  {
+    rcl_time_.nanoseconds = RCL_S_TO_NS(time_msg.sec);
```

Comment:
this overlap the comments in https://github.com/ros2/rcutils/pull/44 but nothing prevents [`time_msg.sec`](https://github.com/ros2/common_interfaces/blob/master/builtin_interfaces/msg/Time.msg#L1) to be negative here. (same everywhere below)